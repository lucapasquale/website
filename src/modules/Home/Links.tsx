import React, { FC } from 'react'
import styled from 'styled-components'

import config from '@src/config'
import { Icon } from '@components/Icon'
import { colors } from '@helpers/theme/colors'

const Section = styled.section({
  maxWidth: '450px',
  marginLeft: '64px',

  '@media (max-width: 700px)': {
    width: '100%',
    marginLeft: '0px',
  },
})

const LinksWrapper = styled.nav({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'space-between',
  marginTop: '16px',

  '@media (max-width: 700px)': {
    display: 'grid',
    gridTemplateColumns: '50% 50%',
  },
})

const IconWrapper = styled.a({
  display: 'flex',
  flexDirection: 'row',
  alignItems: 'center',

  margin: '8px',
  marginBottom: '16px',
  color: `var(${colors.primary.cssVariable})`,

  ':hover': {
    color: `var(${colors.link.cssVariable})`,
  },
})

const LinkLabel = styled.div({
  margin: 0,
  marginLeft: '8px',
  fontSize: '18px',
  fontWeight: 700,
})

export const Links: FC = () => (
  <Section>
    <h3>Contact</h3>

    <LinksWrapper>
      <IconWrapper target="_blank" rel="noopener noreferrer" href={config.LINKS.GITHUB}>
        <Icon name="Github" size="2rem" />

        <LinkLabel>GITHUB</LinkLabel>
      </IconWrapper>

      <IconWrapper target="_blank" href={`mailto:${config.CONTACT.EMAIL}`}>
        <Icon name="Email" size="2rem" />

        <LinkLabel>EMAIL</LinkLabel>
      </IconWrapper>

      <IconWrapper target="_blank" rel="noopener noreferrer" href={config.LINKS.LINKEDIN}>
        <Icon name="Linkedin" size="2rem" />

        <LinkLabel>LINKEDIN</LinkLabel>
      </IconWrapper>

      <IconWrapper target="_blank" rel="noopener noreferrer" href={config.LINKS.INSTAGRAM}>
        <Icon name="Instagram" size="2rem" />

        <LinkLabel>INSTAGRAM</LinkLabel>
      </IconWrapper>
    </LinksWrapper>
  </Section>
)
