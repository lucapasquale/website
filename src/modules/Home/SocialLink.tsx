import React, { FC, ReactNode } from 'react'

type Props = {
  label: string
  url: string
  icon: ReactNode
}

export const SocialLink: FC<Props> = ({ label, url, icon }) => {
  return (
    <a
      target="_blank"
      rel="noopener noreferrer"
      href={url}
      className="text-blue-400 bg-transparent border border-solid border-blue-400 hover:bg-blue-400 hover:text-white active:bg-blue-500 font-bold uppercase text-lg px-4 py-3 flex items-center gap-2 rounded outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
    >
      {icon} {label}
    </a>
  )
}
