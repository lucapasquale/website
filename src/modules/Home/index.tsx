import { Link } from "~components/Link";
import { TechnologyTag } from "~components/TechnologyTag";

export function Page() {
  return (
    <article className="flex flex-col sm:flex-row items-center justify-between space-x-5">
      <section className="container">
        <header>
          <h1 className="scroll-m-20 text-4xl font-extrabold tracking-tight lg:text-5xl">
            Hey, I&apos;m Luca!
          </h1>
        </header>

        <div className="mt-4 text-base sm:text-lg ">
          <p className="leading-7 [&:not(:first-child)]:mt-6 ">
            I&apos;m a software engineer from Brazil, who loves gaming and creating things. I work
            as a full stack engineer at{" "}
            <Link to="https://www.sinai.com" className="font-bold underline">
              Sinai
            </Link>{" "}
            &mdash; where I help decarbonize the planet with{" "}
            <TechnologyTag technology="typescript" /> and <TechnologyTag technology="graphql" />.
          </p>

          <p className="leading-7 [&:not(:first-child)]:mt-6">
            Besides that, I&apos;ve also really enjoy writting <TechnologyTag technology="elixir" />{" "}
            code, and developing games! You can check my last project:{" "}
            <Link to="https://botchini.lucapasquale.dev" className="font-bold underline">
              Botchini
            </Link>
            .
          </p>
        </div>
      </section>

      <section className="flex-shrink-0 mt-4 sm:mt-0 sm:ml-4 aspect-square">
        <img rel="preload" alt="avatar" className="rounded-full " src="/images/avatar.webp" />
      </section>
    </article>
  );
}
