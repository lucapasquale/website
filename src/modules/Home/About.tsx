import Link from 'next/link'
import React, { FC } from 'react'
import { FaGithub, FaEnvelope, FaLinkedin, FaInstagram } from 'react-icons/fa'

import config from '../../config'
import { SocialLink } from './SocialLink'

export const About: FC = () => (
  <article className="max-w-screen-md mx-auto px-4 flex flex-col items-center justify-between gap-4 sm:flex-row sm:items-baseline">
    <section className="max-w-md">
      <h3 className="text-2xl text-gray-300">About me</h3>

      <p>Hey, I'm Luca!</p>
      <p>
        I'm a software engineer born in Brazil, who loves gaming and creating things. I started
        learning programming so I could create my own game, and after that, I never stopped learning
        and improving myself.
      </p>
      <p>
        I've worked mainly on creating APIs and services in GraphQL, either directly for websites,
        or in a micro-service architecture. Nowadays I tend to use TypeScript for developing, but
        I've also used C# and I'm currently learning Elixir.
      </p>
      <p>
        For the frontend, I've created websites, ecommerces and admin dashboards with React, and
        also developed an app using React Native.
      </p>
      <p>
        You can check more about that on my <Link href="/projects">projects page</Link>.
      </p>
    </section>

    <section className="max-w-md">
      <h3 className="text-2xl text-gray-300">Contact</h3>

      <div className="mt-4 px-2 gap-3 grid grid-cols-2 sm:flex sm:flex-col">
        <SocialLink label="GITHUB" url={config.LINKS.GITHUB} icon={<FaGithub />} />
        <SocialLink label="EMAIL" url={`mailto:${config.CONTACT.EMAIL}`} icon={<FaEnvelope />} />
        <SocialLink label="LINKEDIN" url={config.LINKS.LINKEDIN} icon={<FaLinkedin />} />
        <SocialLink label="INSTAGRAM" url={config.LINKS.INSTAGRAM} icon={<FaInstagram />} />
      </div>
    </section>
  </article>
)
