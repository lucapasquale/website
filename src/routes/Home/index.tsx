import React, { FC } from 'react'
import Helmet from 'react-helmet'
import styled from 'styled-components'

import { Layout } from '../../components/Layout'
import { Hero } from '../../components/Hero'
import { About } from './components/About'


const Avatar = styled.img({
  width: '125px',
  height: '125px',
  borderRadius: '50%',
  marginBottom: '16px',
})

export const Home: FC = () => (
  <>
    <Helmet>
      <title>Luca Pasquale</title>
      <link rel="canonical" href="https://lucapasquale.github.io/" />
      <meta name="description" content="My portfolio page" />
    </Helmet>

    <Layout>
      <Hero
        title="Luca Pasquale"
        subTitle="Developer | Gamer | Engineer"
        topContent={<Avatar src="https://avatars2.githubusercontent.com/u/15659967" />}
      />

      <About />
    </Layout>
  </>
)
