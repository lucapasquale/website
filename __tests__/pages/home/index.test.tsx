import { render, screen } from '@testing-library/react'

import Home from '../../../src/pages/index'

describe('Home', () => {
  it('snapshot', () => {
    const { container } = render(<Home />)

    expect(container).toMatchSnapshot()
  })

  it('renders the page', () => {
    render(<Home />)

    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()
    expect(screen.getByRole('img')).toBeInTheDocument()
  })
})
