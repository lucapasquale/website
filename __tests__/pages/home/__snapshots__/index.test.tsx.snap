// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<div>
  <article
    class="flex flex-col sm:flex-row items-center justify-between space-x-5"
  >
    <section
      class="container"
    >
      <header>
        <h1
          class="text-2xl sm:text-4xl font-bold text-gray-200 "
        >
          Hey, I'm Luca!
        </h1>
      </header>
      <div
        class="mt-4 text-base sm:text-lg"
      >
        <p
          class="my-3"
        >
          I'm a software engineer from Brazil, who loves gaming and creating things. I work as a FullStack Engineer at
           
          <a
            class="font-bold underline"
            href="https://www.sinaitechnologies.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Sinai Technologies
          </a>
           
          â€” where I help decarbonize the planet with GraphQL and Typescript.
        </p>
        <p
          class="my-3"
        >
          Besides that, I've also been learning Elixir, and developing some mobile games! You can check my latest one:
           
          <a
            class="font-bold underline"
            href="https://play.google.com/store/apps/details?id=com.Bendev.UpWeGo"
            rel="noopener noreferrer"
            target="_blank"
          >
            Up We Go
          </a>
          .
        </p>
      </div>
    </section>
    <section
      class="flex-shrink-0 mt-4 sm:mt-0 sm:ml-4"
    >
      <img
        alt="test-img"
      />
    </section>
  </article>
</div>
`;
